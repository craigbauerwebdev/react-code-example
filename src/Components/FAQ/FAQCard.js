import React, { useState } from "react";

import ExpandToggle from "Components/Buttons/ExpandToggle";
import OEPAnalytics from "Components/OEPAnalytics";
import faqCardStyles from "./scss/faq-card.module.scss";

const FAQCard = ({ page, answer, question, mobileBorder, index }) => {
  const [expand, setExpand] = useState(false);

  const toggleExpand = () => {
    setExpand(!expand);
  };

  return (
    <div
      className={`${faqCardStyles.faqCard} ${
        mobileBorder && faqCardStyles.mobileBorder
      }`}
    >
      <div className={faqCardStyles.faqItem}>
        <OEPAnalytics
          page={page}
          componentType="Text Link"
          url={question}
          componentName={question}
        >
          <p
            className={`${faqCardStyles.faqQuestion} gtm-FAQ-question`}
            onClick={toggleExpand}
            role="presentation"
          >
            {question}
          </p>
        </OEPAnalytics>
        <ExpandToggle
          expanded={expand}
          page={page}
          handleClick={toggleExpand}
          ariaLabel={[
            "frequently asked question collapse",
            "frequently asked question expand",
          ]}
          ariaControls={`faq-answer-${index}`}
          classList={["faq"]}
        />
      </div>
      <div
        id={`faq-answer-${index}`}
        className={`${faqCardStyles.faqAnswer} ${
          expand && faqCardStyles.expand
        }`}
        aria-hidden={!expand}
      >
        <div
          dangerouslySetInnerHTML={{
            __html: answer,
          }}
        />
      </div>
    </div>
  );
};

export default FAQCard;
