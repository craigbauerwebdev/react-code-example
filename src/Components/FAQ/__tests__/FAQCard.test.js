import FAQCard from "../FAQCard";
import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

jest.mock("Components/Buttons/ExpandToggle", () => "ExpandToggle");

jest.mock("Components/OEPAnalytics", () => "OEPAnalytics");

jest.mock("../scss/faq-card.module.scss", () => ({
  faqCardStyles: () => "",
}));

it("renders without crashing", () => {
  const tree = shallow(<FAQCard page="preevent" />);
  expect(tree.find("OEPAnalytics").get(0).props.page).toBe("preevent");
  expect(toJson(tree)).toMatchSnapshot();
});
