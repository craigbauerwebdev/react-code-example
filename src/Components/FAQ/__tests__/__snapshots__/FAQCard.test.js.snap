// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="undefined undefined"
>
  <div>
    <OEPAnalytics
      componentType="Text Link"
      page="preevent"
    >
      <p
        className="undefined gtm-FAQ-question"
        onClick={[Function]}
        role="presentation"
      />
    </OEPAnalytics>
    <ExpandToggle
      ariaControls="faq-answer-undefined"
      ariaLabel={
        Array [
          "frequently asked question collapse",
          "frequently asked question expand",
        ]
      }
      classList={
        Array [
          "faq",
        ]
      }
      expanded={false}
      handleClick={[Function]}
      page="preevent"
    />
  </div>
  <div
    aria-hidden={true}
    className="undefined false"
    id="faq-answer-undefined"
  >
    <div
      dangerouslySetInnerHTML={
        Object {
          "__html": undefined,
        }
      }
    />
  </div>
</div>
`;
